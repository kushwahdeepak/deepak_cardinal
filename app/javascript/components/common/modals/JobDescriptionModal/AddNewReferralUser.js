import React, { useState } from 'react'
import {
    Col,
    Container,
    Image,
    Row,
    Button,
    Alert,
    Modal,
} from 'react-bootstrap'
import Fab from '@material-ui/core/Fab'
import AddIcon from '@material-ui/icons/Add'
import DeleteIcon from '@material-ui/icons/Delete'
import isEmpty from 'lodash.isempty'

function AddNewReferralUser({
    inviteesDetails,
    handleInputName,
    handleInputEmail,
    handleAddNewRow,
    handleDeleteRow,
    setInvitationMessage,
}) {
    return (
        <Container>
            {inviteesDetails.map((detail, index) => {
                return (
                    <>
                        <Row
                            className="justify-content-md-between align-items-top mt-1"
                            key={index}
                        >
                            <Col
                                xs={5}
                                className="d-flex flex-row align-items-baseline"
                            >
                                <input
                                    type="text"
                                    placeholder="Enter Name"
                                    onChange={(event) => {
                                        handleInputName(event, index)
                                    }}
                                    className="form-control"
                                    value={detail.invitee_name}
                                ></input>
                            </Col>

                            <Col
                                xs={5}
                                className="d-flex flex-row align-items-baseline"
                            >
                                <input
                                    type="text"
                                    placeholder="Enter Email"
                                    onChange={(event) => {
                                        handleInputEmail(event, index)
                                    }}
                                    className="form-control"
                                    value={detail.invitee_email}
                                ></input>
                            </Col>

                            <Col
                                xs={2}
                                className="d-flex flex-row align-items-baseline"
                            >
                                <Fab
                                    size="small"
                                    color="primary"
                                    aria-label="add"
                                >
                                    <AddIcon
                                        onClick={() => handleAddNewRow()}
                                    />
                                </Fab>

                                <Fab
                                    style={{ marginLeft: '10px' }}
                                    size="small"
                                    color="primary"
                                    aria-label="add"
                                >
                                    <DeleteIcon
                                        onClick={() => handleDeleteRow(index)}
                                        size="small"
                                    />
                                </Fab>
                            </Col>
                        </Row>
                    </>
                )
            })}

            <Row className="justify-content-md-between align-items-top mt-2">
                <Col xs={12} className="d-flex flex-row align-items-baseline">
                    <input
                        type="textarea"
                        placeholder="Enter Message"
                        onChange={(event) => {
                            setInvitationMessage(event.target.value)
                        }}
                        className="form-control"
                    ></input>
                </Col>
            </Row>
        </Container>
    )
}

export default AddNewReferralUser
