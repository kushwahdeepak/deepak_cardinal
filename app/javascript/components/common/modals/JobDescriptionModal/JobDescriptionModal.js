import React from "react"
import Modal from 'react-bootstrap/Modal';
import Container from 'react-bootstrap/Container';
import Button from 'react-bootstrap/Button';
import isEmpty from 'lodash.isempty';

function JobDescriptionModal({onHide, show, job}) {

  const hasLocationOrCompany = !isEmpty(job.location) || !isEmpty(job.company)
  if(!job) {
    return (<h2>No suitable job</h2>)
  }

  return (
    <Modal
      onHide={onHide}
      show={show}
      size="lg"
      aria-labelledby="contained-modal-title-vcenter"
      centered
      scrollable
    >
      <Modal.Header closeButton>
        <Modal.Title id="contained-modal-title-vcenter">
          {job.name}
        </Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <Container>
          <p>
            {!isEmpty(job.location) ? job.location + ", " : ""} {job.company} {(hasLocationOrCompany ? " - up to " : "") + job.compensation}
          </p>

          <p>
            {job.description}
          </p>

          <hr/>

          <h5>General requirements</h5>
          <ul>
            {job.genReqs && job.genReqs.length && job.genReqs.map(req => (
              <li style={{display: "list-item", listStyle: "disc"}} key={req}>{req}</li>
            ))}
          </ul>

          <h5>Required skills</h5>
          <ul>
            {job.skills && job.skills.length && job.skills.map(skill => (
              <li style={{display: "list-item", listStyle: "disc"}} key={skill}>{skill}</li>
            ))}
          </ul>

          <h5>Preferred skills</h5>
          <ul>
            {job.prefSkills && job.prefSkills.length && job.prefSkills.map(prefSkill => (
              <li style={{display: "list-item", listStyle: "disc"}} key={prefSkill}>{prefSkill}</li>
            ))}
          </ul>

          <h5>Benefits</h5>
          <ul>
            {job.benefits && job.benefits.length && job.benefits.map(benefit => (
              <li style={{display: "list-item", listStyle: "disc"}} key={benefit}>{benefit}</li>
            ))}
          </ul>
        </Container>
      </Modal.Body>
      <Modal.Footer>
        <Button href={"/jobs/" + job.id + "/edit"} variant="secondary">Edit post</Button>
        <Button type="button" onClick={onHide}>Close</Button>
      </Modal.Footer>
    </Modal>
  );
}

export default JobDescriptionModal
