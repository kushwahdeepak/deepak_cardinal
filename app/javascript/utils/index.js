import { act } from 'react-dom/test-utils'

export const findByTestAttr = (wrapper, attr) => {
    return wrapper.find(`[data-test='${attr}']`)
}

export const timeoutResolve = (time = 0, response = {}) =>
    new Promise((resolve) => {
        setTimeout(() => resolve(response), time)
    })

export const wait = (time = 0) => {
    const timer = timeoutResolve(time)

    return act(async () => {
        await timer
    })
}
export function emailIsValid(email) {
    return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email)
}

export function generateRandom(min, max) {
    return Math.floor(min + Math.random() * (max + 1 - min))
}

export function capitalizeFirstLetter(string) {
    return string.charAt(0).toUpperCase() + string.slice(1).toLowerCase()
}

export function createMarkedList(array) {
    if (!Array.isArray(array)) {
        return []
    }
    return array.map((candidate) => ({
        id: candidate.id,
        name: candidate.id,
        fullName: `${candidate.first_name} ${candidate.last_name}`,
        emailAddress: candidate.email_address,
        isChecked: false,
    }))
}

export function stringReplace(string,toSymbol,fromSymbol){
    return string.replace(toSymbol, fromSymbol)
}

export function capitalize(string) {
    return string.replace(/\b(\w)/g, s => s.toUpperCase());
}

export function skillIsNotEmpty(value){
    return value.trim().length > 0;
}

export function firstCharacterCapital(value){
    return value?.charAt(0)?.toUpperCase() + value?.slice(1)
}
