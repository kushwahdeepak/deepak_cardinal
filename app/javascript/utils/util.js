import isNil from 'lodash.isnil'
import isEmpty from 'lodash.isempty'

class Util {
    /**
     * Define stage values
     */
    static STAGES = ['first_interview', 'second_interview', 'offer', 'reject']

    /**
     * Tests whether a single object attribute includes the at
     * least one string from the given set.
     */
    static attributePassesStringSetFilter(object, attr, stringSet) {
        if (isNil(object[attr])) return false

        // We "OR" the tag tests: the attribute needs to contain only one tag
        return stringSet
            .map((string) => string.trim().toLowerCase())
            .reduce(
                (passes, string) =>
                    passes ||
                    Util.attributeToString(object[attr])
                        .toLowerCase()
                        .includes(string),
                false
            )
    }

    /**
     * Tests whether at least one of a set of object attributes
     * includes every word from a given piece of text.
     */
    static objectPassesTextFilter(object, attrs, text) {
        text = text.trim().toLowerCase()

        if (text === '') return true

        const words = text.split(' ')

        // Join all the given attribute names to lowercase, stringified attribute values
        const allAttrsString = attrs
            .filter((attr) => !isNil(object[attr]))
            .map((attr) => Util.attributeToString(object[attr]).toLowerCase())
            .join('')

        // Is every word included in the allAttrsString?
        return words.reduce(
            (matchesAll, word) => matchesAll && allAttrsString.includes(word),
            true
        )
    }

    static attributeToString(attr) {
        if (typeof attr === 'string') {
            return attr
        } else if (Array.isArray(attr)) {
            return attr.join(',')
        } else if (typeof attr === 'object') {
            return JSON.stringify(attr)
        } else {
            return '' + attr
        }
    }

    static numberWithCommas(x) {
        return x?.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
    }

    /**
     * Returns number of results in format [x] - [y] of [z]
     * @param {number} totalJobsCount Total number of jobs
     * @param {number} pageCount Total number of pages
     * @param {number} activePage Currently active page
     * @param {number} pageSize How many results to display per page
     */
    static displayNumberOfResults(
        totalJobsCount,
        pageCount,
        activePage,
        pageSize,
        totalPersonCount=null,
        candidatePage=false,
        showdisplaycount=true
    ) {
        if (totalJobsCount == 0) return '0 - 0 of 0'
        const offset = pageSize * pageCount - totalJobsCount
        const lastJobIndex =
            (activePage + 1) * pageSize > totalJobsCount
                ? totalJobsCount
                : (activePage + 1) * pageSize
        const firstJobIndex =
            activePage == 0
                ? 1
                : activePage == pageCount - 1
                ? lastJobIndex - pageSize + offset + 1
                : lastJobIndex - pageSize + 1
        return `${firstJobIndex} - ${lastJobIndex} of ${Util.numberWithCommas(
            (candidatePage && showdisplaycount) ? totalPersonCount : totalJobsCount

        )}`
    }

    /**
     * Returns a string indicating the candidate's title and current
     * company, based on available information on the Person object
     * @param {object} candidate Person object from our db
     */
    static candidateCompanyString(candidate) {
        const position = candidate.company_position
        const currentCompany = candidate.current_company

        if (isEmpty(position) && isEmpty(currentCompany)) return ''

        if (!isEmpty(position) && !isEmpty(currentCompany))
            return `${position} at ${currentCompany}`

        if (!isEmpty(position)) return position

        if (!isEmpty(currentCompany)) return `at ${currentCompany}`
    }

    /**
     * Converts the given array of objects with ids to an object whose
     * keys are the ids of the array objects.
     *
     * Every object in the given array must have a unique 'id' property defined.
     *
     * @param {Array} array
     * @returns {Object}
     */
    static arrayToObject(array) {
        const obj = {}
        array.forEach((element) => (obj[element.id] = element))
        return obj
    }

    /**
     * Makes a copy of the given array and replace the value at 'index' with 'value'
     *
     * @param {Array} array
     * @param {Number} index
     * @param {*} value
     * @returns {Array}
     */
    static replaceInArrayCopy_i(array, index, value) {
        const clone = array.slice()
        clone[index] = value
        return clone
    }

    /**
     * Makes a copy of the given array and replaces the element with the same
     * .id property as obj.id with obj
     *
     * @param {Array} array
     * @param {Object} obj
     * @returns {Array}
     */
    static replaceInArrayCopy(array, obj) {
        const index = array.findIndex((element) => element.id === obj.id)
        return Util.replaceInArrayCopy_i(array, index, obj)
    }

    static deleteInArrayCopy(array, index) {
        const copy = array.slice()
        copy.splice(index, 1)

        return copy
    }

    static handleUndefined(value){
        if (value === undefined || value === 'undefined')
        return '';
        else
        return value
    }

    static handleUndefinedFullName(firstName, lastName){
      if(firstName === "undefined" || firstName === undefined)
      return lastName
      else if(lastName === "undefined" || lastName === undefined)
      return firstName
      else
      return firstName + ' ' + lastName     }

    static capitalize(value){
         return value.charAt(0).toUpperCase() + value.slice(1).toLowerCase();
    }
}

export default Util
