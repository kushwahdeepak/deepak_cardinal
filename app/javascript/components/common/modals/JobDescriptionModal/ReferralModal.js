import React, { useState } from 'react'
import axios from 'axios'
import {
    Button,
    Alert,
    Modal,
    Spinner
} from 'react-bootstrap'

import AddNewReferralUser from './AddNewReferralUser'
import validator from 'validator'

function ReferralModal({
    onHide,
    show,
    job,
    setErrorCreatingSubmission,
    setAlertApplyForJob,
}) {
    const [inviteesDetails, setInviteesDetails] = useState([
        { invitee_email: '', invitee_name: '' },
    ])
    const [invitationMessage, setInvitationMessage] = useState('')
    const [errorModel, setErrorModel] = useState(null)
    const [loading, setLoading] = useState(false)

    const handleAddNewRow = () => {
        let temporaryInviteesDetails = inviteesDetails.slice()
        temporaryInviteesDetails.push({ invitee_email: '', invitee_name: '' })
        setInviteesDetails(temporaryInviteesDetails)
    }

    const handleDeleteRow = (index) => {
        let temporaryInviteesDetails = inviteesDetails.slice()
        if (temporaryInviteesDetails.length > 1) {
            temporaryInviteesDetails.splice(index, 1)
            setInviteesDetails(temporaryInviteesDetails)
        }
    }

    const handleInputName = (event, index) => {
        let temporaryInviteesDetails = inviteesDetails.slice()
        temporaryInviteesDetails[index].invitee_name = event.target.value
        setInviteesDetails(temporaryInviteesDetails)
    }

    const handleInputEmail = (event, index) => {
        let temporaryInviteesDetails = inviteesDetails.slice()
        temporaryInviteesDetails[index].invitee_email = event.target.value
        setInviteesDetails(temporaryInviteesDetails)
    }

    const handleInvite = async () => {
        const token = document.querySelector('meta[name="csrf-token"]').content

        let payload = {
            authenticity_token: token,
            job_id: job.id,
        }
        let isError = true;
        payload['referrals'] = inviteesDetails
        payload['message'] = invitationMessage
        inviteesDetails.map(value => {
            if(value.invitee_email === ''){
                setErrorModel("Please Enter Email")
                isError = false
            }
            if (!validator.isEmail(value.invitee_email)) {
                setErrorModel("Please Enter valid Email")
                isError = false
            }
            if(value.invitee_name === ''){
                setErrorModel("Please Enter Name")
                isError = false
            }
        });
        if(invitationMessage === ''){
            setErrorModel("Please Enter Message")
            isError = false
        }
        if(isError){
            setLoading(true)
            try {
                const { data } = await axios.post('/referrals.json', payload)

                if (data.hasOwnProperty('error')) {
                    setErrorCreatingSubmission(data.error)
                    setLoading(false)
                } else {
                    setLoading(false)
                    setInviteesDetails([
                        { invitee_email: '', invitee_name: '' },
                    ])
                    setAlertApplyForJob(
                        'Invited Successfully'
                    )
                }
            } catch (error) {
                setErrorCreatingSubmission(error.message)
                setLoading(false)
            }
            onHide()
        }
    }

    return (
        <Modal
            onHide={onHide}
            show={show}
            size="lg"
            aria-labelledby="contained-modal-title-vcenter"
            centered
            scrollable
        >
            <Modal.Header closeButton>
                <Modal.Title id="contained-modal-title-vcenter">
                    {'Invitations'}
                </Modal.Title>
            </Modal.Header>
            <Modal.Body>
                    {errorModel && (
                            <Alert
                              variant="danger"
                              onClose={() => setErrorModel(null)}
                              dismissible
                            >
                             {errorModel}
                            </Alert>    
                    )}
                    {loading && (
                        <div className="d-flex justify-content-center">
                            <Spinner animation="border" role="status">
                                <span className="sr-only">Loading...</span>
                            </Spinner>
                        </div>
                    )}
                <AddNewReferralUser
                    inviteesDetails={inviteesDetails}
                    handleAddNewRow={handleAddNewRow}
                    handleDeleteRow={(idx) => {
                        handleDeleteRow(idx)
                    }}
                    handleInputName={(event, index) => {
                        handleInputName(event, index)
                    }}
                    handleInputEmail={(event, index) => {
                        handleInputEmail(event, index)
                    }}
                    setInvitationMessage={(data) => {
                        setInvitationMessage(data)
                    }}
                />
            </Modal.Body>
            <Modal.Footer>
                <Button
                    href={'#'}
                    variant="primary"
                    onClick={() => {
                        handleInvite()
                    }}
                >
                    Invite
                </Button>
                <Button type="button"  onClick={() => {
                            onHide()
                        }}variant="secondary">
                    Close
                </Button>
            </Modal.Footer> 
        </Modal>
    )
}

export default ReferralModal
