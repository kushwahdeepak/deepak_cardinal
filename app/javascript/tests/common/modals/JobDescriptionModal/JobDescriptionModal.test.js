import React from 'react';
import JobDescriptionModal from '../../../../components/common/modals/JobDescriptionModal/JobDescriptionModal';
import renderer from 'react-test-renderer';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { findByTestAttr } from '../../../../utils';

configure({adapter: new Adapter()});

const initialShallowRender = (props = {}) => {
	return shallow(<JobDescriptionModal {...props}/>);
};

const initialMountRender = (props = {}) => {
	return mount(<JobDescriptionModal {...props}/>);
};


describe('JobDescriptionModal component',  () => {
	let wrapper;
	let fullWrapper;

	beforeEach(() => {

		const props = {
			job: {location: 'Mock location'}
		};
		wrapper = initialShallowRender(props);
		fullWrapper = initialMountRender(props);
	});

	afterEach(() => {
		fullWrapper.unmount();
	});

	it('should render JobDescriptionModal component', () => {
		expect(wrapper.exists()).toBe(true);
	});

	it('should match with a snapshot', () => {
		const props = {
			job: {location: 'Mock location'}
		};

		const component = renderer.create(<JobDescriptionModal {...props} />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	describe('JobDescriptionModal component with props', () => {

	});
	describe('JobDescriptionModal component without props', () => {

	});

});
