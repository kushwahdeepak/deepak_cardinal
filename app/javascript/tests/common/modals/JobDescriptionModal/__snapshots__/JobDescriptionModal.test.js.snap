// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobDescriptionModal component should match with a snapshot 1`] = `null`;
